### PR TITLE
fetchFromGitorious: add helper

### PR DESCRIPTION
**Pull request by edwtjo**

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -367,12 +367,17 @@ let
 
   fetchzip = import ../build-support/fetchzip { inherit lib fetchurl unzip; };
 
-  fetchFromGitHub = { owner, repo, rev, sha256, name ? "${repo}-${rev}-src" }: fetchzip {
+  fetchFromGitHub = { owner, repo, rev, sha256, name ? "${repo}-${rev}-src", host ? "github.com" }: fetchzip {
```
Comment (nckx) [inline pkgs/top-level/all-packages.nix]: Wouldn't this be more at home in a separate commit? It's more likely to go unnoticed this way.

**Comment by lethalman** [comment]
Honestly calling fetchFromGitHub from fetchFromGitorious is so weird that I'd rather rewrite the implementation. It's so little that it really doesn't need this change. Just call fetchzip like fetchFromGitHub does.

**Comment by edwtjo** [comment]
@lethalman thank you, just what I wanted to hear. That is also how I originally did it in #6293. This PR's modified version was meant to provoke a little :)

**Comment by lethalman** [comment]
I think you can merge it @edwtjo if it works for you?

**Comment by edwtjo** [comment]
closed via 3934bef1384d7e6d6a8df4afb900484101ff67fe
